### DISCUSSION TITLE
Copilot Chat is unable to analyse SQL code

### DISCUSSION THREAD

**Discussion by andreagrandi** [community/community#118670] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Hi,

I'm trying to ask Copilot Chat to simply analyse a SQL query to check if there are issues and instead Copilot Chat is wrapping my code with another Python method, breaking the existing code and not even giving me the answer to the question I asked 🤷🏻‍♂️

Have a code like this:

![Screenshot 2024-04-10 at 08 55 18](https://github.com/community/community/assets/636391/0850556f-4c7e-4ecd-a4e2-caab9e0a6099)

when you hit Enter, instead of answering your question, it will wrap that SQL query with another Python `def upgrade()` method, breaking the existing code and not answering the question:

![Screenshot 2024-04-10 at 08 55 31](https://github.com/community/community/assets/636391/4cbc3011-2ccb-43b3-a145-bb0c6bdcbd73)

Note, this happens always. There is no way to get a valid answer. It also happens in other similar situations: instead of analysing the code, Copilot Chat always breaks the existing code.

Note 2: I've tested this with both the "insiders" version of VSCode (and the preview of the extension) AND with the stable version of VSCode + stable version.

Could you please fix this? Thanks

**NOTE**: I had to use the community forum because your Support bug form here https://support.github.com/contact/bug-report is bugged and doesn't accept the same attachment I've added here 🙄

**Comment by jfullstackdev** [comment]
can you send me those pieces of code ? I will do that on the Chat itself, as for me, it never fails to help even SQL, conversion of it to a seeder or just pure SQL commands

**Comment by eymeen** [comment]
check this answer [#118675](https://github.com/orgs/community/discussions/118675#discussioncomment-9083437)

**Reply by andreagrandi** [reply]
hi, I checked your answer but it's not something that started recently. It has really never worked for me. I remember complaining about this a while ago and I initially though it was because of the insider version of VSCode, but I tried with the stable version too.

**Reply by eymeen** [reply]
I think it's a problem with the copilot extension itself. 
@jdevstatic can you report the copilot team about this bug?
